verl/workers/fsdp_workers.py

```diff
@@ -640,36 +649,103 @@ def compute_ref_log_prob(self, data: DataProto):
     def save_checkpoint(self, local_path, hdfs_path=None, global_step=0, max_ckpt_to_keep=None):
-        # only support save and load ckpt for actor
-        assert self._is_actor
+        assert self._is_actor or self._is_ref
         import torch
 
         if self._is_offload_param:
-            load_fsdp_model_to_gpu(self.actor_module_fsdp)
+            if self._is_actor:
+                load_fsdp_model_to_gpu(self.actor_module_fsdp)
+            elif self._is_ref:
+                load_fsdp_model_to_gpu(self.ref_module_fsdp)
 
         self.checkpoint_manager.save_checkpoint(
             local_path=local_path, hdfs_path=hdfs_path, global_step=global_step, max_ckpt_to_keep=max_ckpt_to_keep
         )
 
         torch.distributed.barrier()
         if self._is_offload_param:
-            offload_fsdp_model_to_cpu(self.actor_module_fsdp)
+            if self._is_actor:
+                offload_fsdp_model_to_cpu(self.actor_module_fsdp)
+            elif self._is_ref:
+                offload_fsdp_model_to_cpu(self.ref_module_fsdp)
 
     @register(dispatch_mode=Dispatch.ONE_TO_ALL)
     def load_checkpoint(self, local_path, hdfs_path=None, del_local_after_load=False):
+        assert self._is_actor or self._is_ref
         if self._is_offload_param:
-            load_fsdp_model_to_gpu(self.actor_module_fsdp)
+            if self._is_actor:
+                load_fsdp_model_to_gpu(self.actor_module_fsdp)
+            elif self._is_ref:
+                load_fsdp_model_to_gpu(self.ref_module_fsdp)
 
         self.checkpoint_manager.load_checkpoint(
             local_path=local_path, hdfs_path=hdfs_path, del_local_after_load=del_local_after_load
         )
 
         if self._is_offload_param:
-            offload_fsdp_model_to_cpu(self.actor_module_fsdp)
+            if self._is_actor:
+                offload_fsdp_model_to_cpu(self.actor_module_fsdp)
+            elif self._is_ref:
+                offload_fsdp_model_to_cpu(self.ref_module_fsdp)
 
-        if self._is_offload_optimizer:
+        if self._is_actor and self._is_offload_optimizer:
             offload_fsdp_optimizer(self.actor_optimizer)
 
+    @register(dispatch_mode=Dispatch.ONE_TO_ALL)
+    def export_actor_weights(self):
+        assert self._is_actor
+
+        from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+        from torch.distributed.fsdp import ShardedStateDictConfig, StateDictType
+
+        if self._is_offload_param:
+            load_fsdp_model_to_gpu(self.actor_module_fsdp)
+
+        # Step 1: Force full state dict mode
+        state_dict_cfg = ShardedStateDictConfig(offload_to_cpu=True)
+
+        with FSDP.state_dict_type(self.actor_module_fsdp, StateDictType.SHARDED_STATE_DICT, state_dict_cfg):
+            model_state_dict = self.actor_module_fsdp.state_dict()
+
+        if self._is_offload_param:
+            offload_fsdp_model_to_cpu(self.actor_module_fsdp)
+
+        return self.rank, model_state_dict
```

Comment:
The model_state_dict can be huge. We can return it directly to the driver. Actually, if we colocate actor and ref in the same worker, we can directly operate on the weight